storage/basestorage.py

```diff
@@ -7,6 +7,10 @@
 logger = logging.getLogger(__name__)
 
 
+class InvalidConfigurationException(Exception):
+    pass
```

Comment:
That's a very generic name. Can we rename this to something more relevant to this particular use case (relevant to storage/cdn spcifically)?